core/controllers/contributor_dashboard.py

```diff
@@ -700,3 +703,190 @@ def post(self):
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {}
+    }
+
+    @acl_decorators.can_fetch_all_contributor_dashboard_stats
+    def get(self, username):
+        """Handles GET requests."""
+
+        user_id = user_services.get_user_id_from_username(username)
+
+        stats = suggestion_services.get_all_contributor_stats(user_id)
+        response = {}
+
+        if stats.translation_contribution_stats is not None:
+            response['translation_contribution_stats'] = _get_complete_stats(
+                stats.translation_contribution_stats, True)
+
+        if stats.translation_review_stats is not None:
+            response['translation_review_stats'] = _get_complete_stats(
+                stats.translation_review_stats, False)
+
+        if stats.question_contribution_stats is not None:
+            response['question_contribution_stats'] = _get_complete_stats(
+                stats.question_contribution_stats, False)
+
+        if stats.question_review_stats is not None:
+            response['question_review_stats'] = _get_complete_stats(
+                stats.question_review_stats, False)
+
+        self.render_json(response)
+
+
+def _get_complete_stats(
+    stats_data: List[Union[
+        suggestion_registry.TranslationContributionStats,
+        suggestion_registry.TranslationReviewStats,
+        suggestion_registry.QuestionContributionStats,
+        suggestion_registry.QuestionReviewStats
+    ]],
+    stats_are_translation_contribution
+):
+    """Returns corresponding stats dicts with all the necessary
+    information for the frontend.
+
+    Args:
+        stats_data: list. Stats domain objects.
+        stats_are_translation_contribution: bool. A flag that indicates whether
+            stats are translation contributions.
+
+    Returns:
+        list. Dict representations of TranslationContributionStats/
+        TranslationReviewStats/QuestionContributionStats/
+        QuestionReviewStats domain objects with additional keys:
+            topic_name: str. Topic name.
+            contribution_months: str. Unique translation contribution
+                months of format: "%b %Y", e.g. "Jan 2021".
+        Unnecessary keys topic_id, contribution_dates, contributor_user_id
+        are consequently deleted.
+    """
+    translation_review_stats_dicts = [
+        stats.to_dict() for stats in stats_data
+    ]
+    topic_ids = [
+        stats_dict['topic_id']
+        for stats_dict in translation_review_stats_dicts
+    ]
+    topic_summaries = topic_fetchers.get_multi_topic_summaries(topic_ids)
+    topic_name_by_topic_id = {}
+    for topic_summary in topic_summaries:
+        if topic_summary is None:
+            continue
+        topic_name_by_topic_id[topic_summary.id] = topic_summary.name
+    for stats_dict in translation_review_stats_dicts:
+        stats_dict['topic_name'] = topic_name_by_topic_id.get(
+            stats_dict['topic_id'], 'UNKNOWN')
+
+        if stats_are_translation_contribution:
```

Comment:
Instead of passing in this boolean everywhere, can we just check for the existence of the `first_contribution_date` and `last_contribution_date` keys?

Also, can you add a TODO to migrate TranslationContributionStats to use `first_contribution_date` and `last_contribution_date` instead?

Reply:
Done